ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Storage/ChainDB.hs

```diff
@@ -38,11 +38,48 @@
 --   chain. 'Ouroboros.Consensus.Storage.ChainDB.API.ChainDB' defines the chain
 --   DB API.
 --
--- NOTE: at the moment there is an inconsistency in the module structure for
--- each of these components. In particular,
--- "Ouroboros.Consensus.Storage.LedgerDB" contains the whole definition and API
--- for the LedgerDB, but the other three databases are broken up into multiple
--- smaller submodules. We aim to resolve this when UTxO-HD is merged.
+-- == Resource management in the ChainDB
+--
+-- Each of the databases in the ChainDB can produce resources that need to be
+-- eventually freed. In particular:
+--
+-- - The LedgerDB is used to create 'Forker's
+--
+-- - The ChainDB is used to create 'Follower's (which in turn contain
+--   'Iterator's).
+--
+-- The 'runWith' function in Consensus spawns a resource registry (which we will
+-- name __the consensus registry__) which will contain the ChainDB. Shutting
+-- down the Consensus layer is what will close the ChainDB by the consensus
+-- resource registry going out of scope.
```

Comment:
```suggestion
-- name __the consensus registry__) which will contain the ChainDB. Shutting
-- down the Consensus layer will close the ChainDB because of the consensus
-- resource registry goes out of scope.
```